packages/ember-model/lib/belongs_to.js

```diff
@@ -96,7 +96,9 @@ Ember.belongsTo = function(type, options) {
 
       if (meta.options.embedded) {
         if (oldValue) {
-          oldValue.removeObserver('isDirty', dirtyChanged);
+          try {
+            oldValue.removeObserver('isDirty', dirtyChanged);
+          } catch {}
```

Comment:
Adding a `hasObserverFor` guard doesn't seem to prevent this from raising exceptions. I'll need to dig into this a little more soon